secedgar/filings/company.py

```diff
@@ -11,37 +11,41 @@
 from secedgar.utils import sanitize_date
 
 
-class Filing(AbstractFiling):
+class CompanyFilings(AbstractFiling):
     """Base class for receiving EDGAR filings.
 
     Args:
         cik_lookup (str): Central Index Key (CIK) for company of interest.
-        filing_type (secedgar.filings.filing_types.FilingType): Valid filing type enum.
-        start_date (Union[str, datetime.datetime], optional): Date before which not to
-            fetch reports. Stands for "date after."
-            Defaults to None (will fetch all filings before end_date).
-        end_date (Union[str, datetime.datetime], optional): Date after which not to fetch reports.
+        filing_type (Union[secedgar.filings.filing_types.FilingType, None]): Valid filing type
+            enum. Defaults to None. If None, then all filing types for CIKs will be returned.
+        start_date (Union[str, datetime.datetime, datetime.date], optional): Date before
+            which not to fetch reports. Stands for "date after."
+            Defaults to None (will fetch all filings before ``end_date``).
+        end_date (Union[str, datetime.datetime, datetime.date], optional):
+            Date after which not to fetch reports.
             Stands for "date before." Defaults to today.
         count (int): Number of filings to fetch. Will fetch up to `count` if that many filings
             are available. Defaults to all filings available.
-        kwargs: See kwargs accepted for :class:`secedgar.client.network_client.NetworkClient`.
+        kwargs: See kwargs accepted for :class:`secedgar.client.NetworkClient`.
 
     .. versionadded:: 0.1.5
     """
 
     def __init__(self,
                  cik_lookup,
-                 filing_type,
+                 filing_type=None,
                  start_date=None,
-                 end_date=datetime.datetime.today(),
+                 end_date=date.today(),
                  client=None,
                  count=None,
+                 ownership='include',
+                 match_format='ALL',
                  **kwargs):
         # Leave params before other setters
         self._params = {
             'action': 'getcompany',
             'output': 'xml',
-            'owner': 'include',
+            'owner': ownership,
```

Comment:
I don't think that this quite solved the issue mentioned in #179. For example, I think that both 424H and 424H/A filings are ownership filings. However, the way we are currently searching, if one were to only search for 424H, the 424H/A filings would also show up.

I think what we need to do is have a check in `_get_urls_for_cik`.

Reply:
That particular bug is fixed by this commit: https://github.com/sec-edgar/sec-edgar/pull/174/commits/7f5462a277045f667b9c77d3aedc5f046ac8dfdb